experimental/dds/tree2/src/feature-libraries/simple-tree/proxies.ts

```diff
@@ -614,12 +646,23 @@ const mapStaticDispatchMap: PropertyDescriptorMap = {
 			key: string,
 			value: ProxyNodeUnion<AllowedTypes, "javaScript">,
 		): SharedTreeMap<MapNodeSchema> {
+			const node = getEditNode(this);
+
 			const { content, hydrateProxies } = extractFactoryContent(
 				value as FlexibleFieldContent<MapFieldSchema>,
 			);
+			const mappedContent =
+				content === undefined
+					? undefined
+					: cursorFromProxyTree(
+							content,
+							node.context,
+							node.schema.mapFields?.types ??
```

Comment:
After the my latest PR that you reviewed, `mapFields` should be guaranteed to be present here now (since `node.schema` is now strongly typed as a MapNodeSchema)